src/compiler/scala/tools/nsc/classpath/AggregateClassPath.scala

```diff
@@ -41,22 +41,36 @@ case class AggregateClassPath(aggregates: Seq[ClassPath]) extends ClassPath {
 
   // This method is performance sensitive as it is used by SBT's ExtractDependencies phase.
   override def findClass(className: String): Option[ClassRepresentation] = {
-    val (pkg, simpleClassName) = PackageNameUtils.separatePkgAndClassNames(className)
-
-    def findEntry(isSource: Boolean): Option[ClassRepresentation] = {
-      aggregatesForPackage(PackageName(pkg)).iterator.map(_.findClass(className)).collectFirst {
-        case Some(s: SourceFileEntry) if isSource => s
-        case Some(s: ClassFileEntry) if !isSource => s
-      }
+    // workaround a performance bug in exiting versions of Zinc.
+    // https://github.com/sbt/zinc/issues/757
+    // Retire this code once the Zinc releases a fix and it is widely used.
+    val noByteCode = className match {
+      case s if s.indexOf('<') >= 0 => true
```

Comment:
What's this one matching?

Reply:
Oh: `<byname>`, `<repeated>`, etc..

Reply:
... which currently come out of Zinc as `"<byname>assert"` or ``"<repeated>$treeCreato"` and the like, due to a bug in `fullName` in Zinc for names with special characters!